workshops/meme-generator/README.md

```diff
@@ -0,0 +1,385 @@
+    ```
+    
+    (put this after our `<h1>` tag)
+
+* Next, we'll need to take a meme template as a file input, we'll use:
+  
+  ```html
+  <input type="file" id="image-input" accept="image/*">
+  ```
+
+    This will pop-up a button on our website that'll let the user input a meme template as an image.
+
+* Now we need a generate button:
+
+    ```html
+    <button id="generate-btn">Generate!</button>
+    ```
+
+    Later we can add an event listener to this button which can run a function on click!
+
+* But we need something to display our meme, right?
+
+    For this, we'll use an [HTML5 Canvas](https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API):
+
+    ```html
+    <canvas id="meme-canvas" title="Right click to save this meme"></canvas>
+    ```
+
+Our final HTML file will look like:
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="utf-8">
+        <title>Meme Generator</title>
+        <link rel="stylesheet" href="style.css">
+    </head>
+    <body>
+        <h1>Welcome to Meme-Genie 🧞 </h1>
+        <div>
+            <textarea id="top-text"></textarea>
+            Text size: <input type="range" id="top-text-size-input" min="0.05" max="0.25" value="0.15" step="0.01">
+        </div>
+        <div>
+            <textarea id="bottom-text"></textarea>
+            Text size: <input type="range" id="bottom-text-size-input" min="0.05" max="0.25" value="0.15" step="0.01">
+        </div>
+        <div>
+            <input type="file" id="image-input" accept="image/*">
+        </div>
+        <div>
+            <button id="generate-btn">Generate!</button>
+        </div>
+        <div>
+            <canvas id="meme-canvas" title="Right click to save this meme"></canvas>
+        </div>
+        <script src="script.js"></script>
+    </body>
+</html>
+```
+
+Note that we have added the *Text Size* slider which we'll use to increase or decrease the text size.
+You can take a look at how different input types work [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input).
+
+### 2) CSS
+
+Next, let's add some basic styling to our website, navigate to our `style.css` file and add the following code:
+
+```CSS
+h1 {
+  font-family: Impact, 'Arial Narrow Bold', sans-serif;
+  font-size: 30px;
+}
+
+body {
+  margin: 10px;
+  background-color: lightblue;
+}
+
+#meme-canvas {
+  width: 300px;
+}
+```
+
+So in this CSS code, we first change our heading's font and font size, then we apply a background color to our body, and finally we select the meme canvas by it's ID and set its width to 300 pixels.
+
+That's it! This is all the CSS we'll be doing in the workshop.
+
+### 3) JavaScript
+
+Now, navigate to our `script.js` file and:
+
+* Make a `function` called `generateMeme`, which will take: `image`, `topText`, `bottomText`, `topTextSize`, `bottomTextSize` variables as the function arguments.
+  
+  Here's how our `generateMeme` function will look like:
```

Comment:
Instead of showing the whole function, add a separate instruction(s) with each of the steps. You should also break this function down into separate steps, because this is a really big code block.